Sources/MongoSwift/SDAM.swift

```diff
@@ -353,4 +375,65 @@ public struct TopologyDescription: Equatable {
         self.servers = size > 0 ? Array(buffer).map { ServerDescription($0!) } : []
         // the buffer is documented as always containing non-nil pointers (if non-empty).
     }
+
+    // For testing purposes
+    internal init(type: TopologyType, servers: [ServerDescription]) {
+        self.type = type
+        self.servers = servers
+    }
+}
+
+extension TopologyDescription {
+    internal func findSuitableServers(readPreference: ReadPreference? = nil) -> [ServerDescription] {
+        switch self.type {
```

Comment:
so the reason this top level switch needs the `default` case is because `TopologyType` is a struct rather than an enum, so there is no way to exhaustively switch on possible `TopologyType`s. 

we did this so that we could add new topology types to the public API (such as `.loadBalanced`, most recently) without making what is considered a breaking API change. currently in Swift adding a new case to an enum is considered breaking, because a `switch` statement that was previously considered exhaustive (no `default` case required) is no longer exhaustive, so if a user was switching on that type their code may no longer compile.

under the hood though, `TopologyType` has an internal `_TopologyType` enum which does provide an exhaustive list of possible valid topology types. it's ok to use an enum for the internal representation since that isn't exposed to users, and if we add a case to it we can also just update any `switch`s we have on it internally at the same time, and nothing breaks.

internally, we should prefer to exhaustively switch on these internal representations since we can then explicitly handle each case, and then if we add a new `TopologyType` (and corresponding `_TopologyType`) the switches will stop compiling and we'll know we need to update them to handle whatever the new topology type is. (as written this would continue to compile but just always return no servers for a new topology type).

so I think it would be a little better to switch on `self.type._topologyType` here and then you can remove the `default` case. to do that, I think you'll need to make `_TopologyType` and `TopologyType._topologyType` `fileprivate`, they're currently `private`

Reply:
Ah thank you for the explanation! I made the changes